File: 0003_Longest Substring Without Repeating Characters/LongestSubstringWithoutRepeatingCharacters_2.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var arr = new int[128];
        var l = 0;
        var r = 0;
        var ans = 0;
        while(r < s.Length)
        {
            if(arr[s[r]] == 0)
            {
                arr[s[r]]++;
                ans = Math.Max(ans, r-l+1);
            }else{
                while(s[l] != s[r])
                {
                    arr[s[l++]]--;
                }
                l++;
            }
            
            r++;
        }
        
        return ans;
    }
}

File: 0003_Longest Substring Without Repeating Characters/LongestSubstringWithoutRepeatingCharacters_2.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int l = 0;
        int ans = 0;
        for(int i=0;i<s.length();i++) {
            while(!set.add(s.charAt(i))) set.remove(s.charAt(l++));
            ans = Math.max(ans, i - l + 1);
        }
        
        return ans;
    }
}